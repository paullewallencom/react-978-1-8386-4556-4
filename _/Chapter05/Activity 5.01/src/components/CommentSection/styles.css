.comments {
  margin: 5% 0;
}
