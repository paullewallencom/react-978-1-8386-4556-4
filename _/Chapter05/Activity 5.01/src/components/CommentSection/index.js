import React, { Component } from "react";
// styles
import "./styles.css";
// components
import { Comment } from "../Comment";

class CommentSection extends Component {
  render() {
    const { comments } = this.props;
    if (!comments || comments.length === 0) {
      return null;
    }
    return (
      <section className="comments">
        <h1>Comments</h1>
        {comments.map((comment, key) => (
          <Comment key={`comment_${key}`} comment={comment} level={1} />
        ))}
      </section>
    );
  }
}

export default CommentSection;
