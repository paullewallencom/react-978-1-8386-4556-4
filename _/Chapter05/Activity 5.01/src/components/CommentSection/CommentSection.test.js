import React from "react";
import { render, cleanup } from "@testing-library/react";
import CommentSection from "./index";

afterEach(cleanup);

it("renders CommentSection", () => {
  const { asFragment } = render(<CommentSection />);
  expect(asFragment()).toMatchSnapshot();
});

it("renders nothing when comment is not defined", () => {
  const { container } = render(<CommentSection />);
  expect(container.firstChild).toBeNull();
});

it("renders nothing when empty comment", () => {
  const { container } = render(<CommentSection comments={[]} />);
  expect(container.firstChild).toBeNull();
});

const comments = [
  {
    name: "Karen Michaels",
    text:
      "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean scelerisque, purus ac feugiat eleifend, ex.",
    image:
      "https://images.unsplash.com/photo-1567336273898-ebbf9eb3c3bf?auto=format&fit=crop&w=200&q=80",
    time: "Oct 10, 2019"
  },
  {
    name: "Tim Parker",
    text:
      "Tullam elementum vulputate lectus sed pellentesque. Maecenas convallis orci at elit consectetur tempus ac id.",
    image:
      "https://images.unsplash.com/photo-1505247964246-1f0a90443c36?auto=format&fit=crop&w=200&q=80",
    time: "Oct 05, 2019"
  }
];

it("renders 2 comments when they are provided", () => {
  const { container } = render(<CommentSection comments={comments} />);
  expect(container.querySelectorAll("div.comment").length).toBe(2);
});
