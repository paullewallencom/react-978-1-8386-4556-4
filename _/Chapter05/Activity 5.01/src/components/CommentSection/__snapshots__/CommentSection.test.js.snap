// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CommentSection 1`] = `<DocumentFragment />`;
